### ISSUE TITLE
Clearing the text field

### ISSUE THREAD

**Issue by ShayanArifButt** [merlinthemagic/MTS#23]
How can we clear the text field ?  so i was trying to implement captcha solve by user, but if i save the window object and then enter the captcha , or corrected email by user ,  it actually appends to the text field , i assume this is because i serialized and saved the previous state of the page and hence it already contained the email i entered before; here is the image;

![image](https://user-images.githubusercontent.com/28813039/30125372-a8d88512-9351-11e7-9def-2faae5bb3936.png)

you can see in the email field , that emails are appended to the previous state email.

In short i need something like clear text field , then i will be able to clear the field and enter the posted email  etc

**Comment by merlinthemagic** [comment]
You appear to know JS quite well. How about pushing a script that sets element.val() to ""? Another option is sending a bunch of "backspace" keys...

**Comment by angular013** [comment]
@merlinthemagic 

how can we inject our javascript or jquery code into the page ? by pushing the script , you meant injecting our script ,right ?

**Comment by ShayanArifButt** [comment]
@merlinthemagic 

can we inject jquery code as well ?

**Comment by merlinthemagic** [comment]
If the JQuery framework is loaded, sure you can inject a script.

See documentation for the windows methods:

loadJS
callJSFunction

**Comment by merlinthemagic** [comment]
@angular013 can i go ahead and close?

**Comment by ShayanArifButt** [comment]
@merlinthemagic 

sure the javascript function inject is working fine.

`$clear_fields	= "function clear_fields(){
						
						document.getElementById('address-ui-widgets-enterAddressFullName').value='';
						document.getElementById('address-ui-widgets-enterAddressLine1').value='';
						document.getElementById('address-ui-widgets-enterAddressCity').value='';
						document.getElementById('address-ui-widgets-enterAddressStateOrRegion').value='';
						document.getElementById('address-ui-widgets-enterAddressPostalCode').value='';
						document.getElementById('address-ui-widgets-enterAddressPhoneNumber').value='';
						
						
						}";
								
					$windowObj->loadJS($clear_fields);					
					$funcReturn = $windowObj->callJSFunction('clear_fields');`

although a question , can we pass parameter in the function and if we want to get a return value then how will we go about doing that ? , eg passing parameter something like this

`$funcReturn = $windowObj->callJSFunction('clear_fields($parameter)');``

because few days ago i tried to do something like that but it gave some error if i remember correctly

**Comment by merlinthemagic** [comment]
no. you can only pass the function name. But how about setting a global variable that is used inside the function? then set that variable each time and then call the function?

**Comment by ShayanArifButt** [comment]
@merlinthemagic 

ok thanks , will try that . you can close this issue
